src/App/Fossa/Telemetry/Types.hs

```diff
@@ -0,0 +1,122 @@
+module App.Fossa.Telemetry.Types (
+  CliEnvironment (..),
+  SystemInfo (..),
+  TelemetryRecord (..),
+  TimedLogRecord (..),
+  TelemetrySink (..),
+  TelemetryCtx (..),
+  TelemetryTimeSpent (..),
+  TelemetryCmdConfig (..),
+  CountableCliFeature (..),
+) where
+
+import Control.Concurrent.STM (TMVar)
+import Control.Concurrent.STM.TBMQueue (TBMQueue)
+import Data.Aeson
+import Data.Aeson.Types (toJSONKeyText)
+import Data.Map (Map)
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Time (UTCTime)
+import Data.Tracing.Instrument (CounterRegistry)
+import Data.UUID (UUID)
+import Effect.Logger (Severity)
+import Fossa.API.Types (ApiOpts)
+import GHC.Generics (Generic)
+
+data TelemetrySink
+  = TelemetrySinkToEndpoint ApiOpts
+  | TelemetrySinkToFile
+  deriving (Show, Eq, Ord)
+
+data TelemetryCtx = TelemetryCtx
+  { telId :: UUID
+  , telLogsQ :: TBMQueue TimedLogRecord
+  , telSink :: TMVar TelemetrySink
+  , telTimeSpent :: TBMQueue TelemetryTimeSpent
+  , telFossaConfig :: TMVar (Text, Value)
+  , telCounters :: CounterRegistry CountableCliFeature Int
+  , telStartUtcTime :: UTCTime
```

Comment:
Can we reorder these fields?  I'd prefer to see a pattern of either sorting by datatype (similar types grouped), or alphabetically by field accessor.  I think we should make this a firm rule for all large datatypes like this one.